experiment_helper/cli_parser.py

```diff
@@ -0,0 +1,119 @@
+from typing import Literal
+
+import torch
+from pydantic import Field, AliasChoices
+from pydantic_settings import BaseSettings, CliImplicitFlag
+from experiment_helper.experiment_typing import (
+    LargeModel,
+    ModelDtype,
+)
+from cezo_fl.random_gradient_estimator import RandomGradEstimateMethod
+from cezo_fl.util.language_utils import SUPPORTED_LLM
+from experiment_helper.data import DataSetting  # noqa: F401
+
+
+class GeneralSetting(BaseSettings, cli_parse_args=True):
```

Comment:
is there an easy to list all flags?

Reply:
I do not fully understand the question

Reply:
we want to have a list to tell how many settings there are?

something like this?:
[GeneralSetting, DeviceSetting, ModelSetting, ...]

Reply:
Will add some docstring in the future when we are going to make this repo a library, need to give some documentation for user